lib/Status/Listener.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2019 Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Status;
+
+use OCA\Talk\Events\ModifyParticipantEvent;
+use OCA\Talk\Room;
+use OCA\Talk\Signaling\Messages;
+use OCA\UserStatus\Service\PredefinedStatusService;
+use OCP\EventDispatcher\IEventDispatcher;
+use OCP\IUserSession;
+use OCP\UserStatus\IManager;
+use OCP\UserStatus\IUserStatus;
+
+class Listener {
+	/** @var IUserSession */
+	public $userSession;
+
+	public function __construct(IUserSession $userSession) {
+		$this->userSession = $userSession;
+	}
+
+	public static function register(IEventDispatcher $dispatcher): void {
+		$dispatcher->addListener(Room::EVENT_BEFORE_SESSION_JOIN_CALL, static function (ModifyParticipantEvent $event) {
+			$room = $event->getRoom();
+			/** @var \OCA\Talk\Chat\SystemMessage\Listener $listener */
+			$listener = \OC::$server->get(self::class);
+			/** @var IManager $statusManager */
+			$statusManager = \OC::$server->get(IManager::class);
+
+			$statusManager->setUserStatus($listener->userSession->getUser()->getUID(), PredefinedStatusService::CALL, IUserStatus::DND, true);
```

Comment:
As said in the past this is a very crucial and fatal automation which at least needs an opt-out.

If you are on dnd, you will not receive push notifications to your mobile apps anymore. Since now i would be automatically on dnd all the time I basically would never receive push notifications.

Also not every call is like a meeting.

Reply:
When discussing the feature with @jancborchardt and @nimisha-vijay we decided to keep it simple and not make it configurable.

Maybe something we could change instead of setting a DND status is to not change the current status and just change the message. DND is anyway often already enabled at the OS level when sharing a screen for example and I could see how not receiving notifications during every meeting could be annoying.

> Also not every call is like a meeting.

Trying to detect that type of call is currently happening is hard and probably not a good idea in terms of privacy since every user can see it.

Reply:
> Trying to detect that type of call is currently happening is hard and probably not a good idea in terms of privacy since every user can see it.

We will soon have such a distinction (see the roadmap project) to pick certain other settings too, e.g. if everyone can unmute or moderators have to give permissions etc.

Maybe we can discuss it next week when I'm back

Reply:
@jancborchardt @CarlSchwan Shall we have a call about it on monday?

Reply:
Yes, I'm free for a call from 13h to 17h on Monday